integrations/server/test_api_keys.py

```diff
@@ -0,0 +1,243 @@
+ 
+    def test_public_route(self):
+        """Test public route"""
+        public_route = "http://delphi_web_epidata/epidata/version"
+        status_codes = set()
+        for _ in range(10):
+            status_codes.add(self._make_request(public_route).status_code)
+        self.assertEqual(len(status_codes), 1)
+        self.assertEqual(next(iter(status_codes)), 200)
+
+    def test_no_multiples_data_source(self):
+        """Test requests with no multiples and with provided `data_source` and `signal` as a separate query params."""
+        params = {
+            "source": "covidcast",
+            "data_source": "fb-survey",
+            "signal": "smoothed_wcli",
+            "time_type": "day",
+            "geo_type": "state",
+            "geo_value": "pa",
+            "time_values": "20200406",
+        }
+        status_codes = set()
+        for _ in range(10):
+            status_codes.add(self._make_request(params=params).status_code)
+        self.assertEqual(len(status_codes), 1)
+        self.assertEqual(next(iter(status_codes)), 200)
+
+    def test_no_multiples_source_signal(self):
+        """Test requests with colon-delimited source-signal param presentation."""
+        params = {
+            "source": "covidcast",
+            "signal": "fb-survey:smoothed_wcli",
+            "time_type": "day",
+            "geo_type": "state",
+            "geo_value": "pa",
+            "time_values": "20200406",
+        }
+        status_codes = set()
+        for _ in range(10):
+            status_codes.add(self._make_request(params=params).status_code)
+        self.assertEqual(len(status_codes), 1)
+        self.assertEqual(next(iter(status_codes)), 200)
+
+    def test_multiples_allowed_signal(self):
+        """Test requests with 2 multiples and allowed dashboard signal"""
+        params = {
+            "source": "covidcast",
+            "signal": "fb-survey:smoothed_wcli",
+            "time_type": "day",
+            "geo_type": "state",
+            "geo_value": "pa,ny",
+            "time_values": "20200406,20200407",
+        }
+        status_codes = set()
+        for _ in range(10):
+            status_codes.add(self._make_request(params=params).status_code)
+        self.assertEqual(len(status_codes), 1)
+        self.assertEqual(next(iter(status_codes)), 200)
+
+    def test_multiples_non_allowed_signal(self):
+        """Test requests with 2 multiples and non-allowed dashboard signal"""
+        params = {
+            "source": "covidcast",
+            "signal": "hospital-admissions:smoothed_adj_covid19_from_claims",
+            "time_type": "day",
+            "geo_type": "state",
+            "geo_value": "pa,ny",
+            "time_values": "20200406,20200407",
+        }
+        status_codes = set()
+        for _ in range(10):
+            status_codes.add(self._make_request(params=params).status_code)
+        self.assertEqual(len(status_codes), 2)
+        self.assertEqual(status_codes, {200, 429})
```

Comment:
```suggestion
        self.assertEqual(status_codes, {200, 429})
```
second line makes the first redundant

(more similar usages of this below too)